Enterprise-CMCS/managed-care-review
# MR-3173: State indicates the population served by contract action

## Summary
[MR-3173](https://qmacbis.atlassian.net/browse/MR-3173)

State users can indicate the population covered by the contract. CMS at a glance can know what population the contract covers. 

- Add `populationCovered` field to protobuf schema
- Updated helper functions, data mocks, and GraphQL schema.
- Add population covered question to `Submission Type`, `Review and Submit`, and `Submission Summary` pages.
- Add 'chip-only-form` feature flag.
- Hide UI and validations on the front end and back end behind the feature flag.
- Update `submitHealthPlanPackage` and `createHealthPlanPackage` resolvers for `populationCovered`.

#### Related issues

#### Screenshots
This UI change to the `Submission Type` field has been changed from the Figma design. Please see this [Slack](https://cmsgov.slack.com/archives/C014CRU197U/p1680533769865229) thread for the discussion

<img src="https://user-images.githubusercontent.com/98117700/229874131-928acae9-02d1-4347-a341-fde5701f0c46.png" width="400" />

#### Test cases covered

Unit Tests
- `submitHealthPlanPackage.test.ts`
   - `'errors when population coverage question is undefined'`
- `SubmissionTypeSummarySection.test.ts`
   - `'renders expected fields for draft package on review and submit'`
      - Add test for population coverage question
   - `'renders missing field message for population coverage question when expected'`
- `SubmissionType.test.tsx`
   - `'displays population coverage question'`
   - `'disables contract and rates submission type radio and displays hint when CHIP only is selected'`
   - `'switches submission type to contract only when changing existing population coverage to CHIP-only'`
   - `'new submissions does not automatically select contract only submission type when selecting CHIP-only coverage'`
   - `'does not clear contract only submission type radio when switching to CHIP-only population coverage'`
   - `'shows validation message when population coverage is not selected'`

Cypress Tests
- `stateSubmissionFormCommands.ts`
   - Add code to select a population coverage radio in each command.
      - `'fillOutContractActionOnlyWithBaseContract'`
      - `'fillOutContractActionOnlyWithAmendment'`
      - `'fillOutContractActionAndRateCertification'`
-`launchDarklyCommands.ts`
   - `'stubFeatureFlags'` command defaults `chip-only-form` to true
   - This will toggle on the `chip-only-form` flag everywhere that `'stubFeatureFlags'`, reducing the need to `cy. interceptFeatureFlags` in every spec.
- `questionResponse.spec.ts`
   - Add `chip-only-form` to feature flag interception, because calling `cy.interceptFeatureFlags` overwrites the default we set in `'stubFeatureFlags'`. I would keep this behavior to force intercept commands to start tests with flags at a default state and force the user to know exactly what flags are turned on the for test.






## QA guidance

Just a quick tip for testing old data with the new question. You can toggle the flag off, fill out a submission completely, and then toggle the flag on.
